modules/host.go

```diff
@@ -4,29 +4,54 @@ import (
 	"net"
 
 	"github.com/NebulousLabs/Sia/consensus"
+	"github.com/NebulousLabs/Sia/network"
 )
 
 const (
 	AcceptContractResponse = "accept"
 )
 
+// ContractTerms are the parameters agreed upon by a client and a host when
+// forming a FileContract.
+type ContractTerms struct {
+	FileSize           uint64
+	StartHeight        consensus.BlockHeight
+	WindowSize         consensus.BlockHeight // how many blocks a host has to submit each proof
+	NumWindows         uint64
+	Price              consensus.Currency // client contribution towards payout each window
+	Collateral         consensus.Currency // host contribution towards payout each window
+	ValidProofAddress  consensus.CoinAddress
+	MissedProofAddress consensus.CoinAddress
+}
+
+type HostInfo struct {
+	HostSettings
+
+	StorageRemaining int64
+	NumContracts     int
+}
+
 type Host interface {
-	// Announce puts an annoucement out so that clients can find the host.
-	AnnounceHost(freezeVolume consensus.Currency, freezeUnlockHeight consensus.BlockHeight) (consensus.Transaction, error)
+	// Announce announces the host on the blockchain. A host announcement
+	// requires two things: the host's address, and a volume of "frozen"
+	// (time-locked) coins, used to mitigate Sybil attacks.
+	Announce(network.Address, consensus.Currency, consensus.BlockHeight) error
```

Comment:
names might help document stuff in this case. It's not really clear that the BlockHeight corresponds to when you want the coins unfrozen, or that the act of calling Announce is actually going to freeze your coins.